src/backend/TxCache.cpp

```diff
@@ -0,0 +1,105 @@
+#include <backend/TxCache.h>
+namespace Backend {
+
+uint32_t
+TxCache::latestLedgerSequence() const
+{
+    std::shared_lock lck{mtx_};
+    return latestSeq_;
+}
+
+// adds an entire ledger's worth of transactions to the cache. Evicts oldest
+// ledger.
+void
+TxCache::update(
+    std::vector<ripple::uint256> const& hashes,
+    std::vector<TransactionAndMetadata> const& transactions,
```

Comment:
We probably want to `assert` that `transactions` and `hashes` are the same length